### PR TITLE
Speed dial search implemented

### PR DESCRIPTION
**Pull request by Merkost**

### CODE REVIEW

File: app/src/main/res/values-ar/strings.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <string name="allow_displaying_over_other_apps">الرجاء السماح بالعرض فوق التطبيقات الأخرى للحصول على سلوك موثوق</string>
     <!-- Contacts -->
     <string name="could_not_access_contacts">تعذر الوصول إلى جهات الاتصال</string>
+    <string name="select_contact">Select contact</string>
```
Comment (tibbi) [inline app/src/main/res/values-ar/strings.xml:9]: theres already a Choose contact string in Simple Commons, just use it

File: app/src/main/kotlin/com/simplemobiletools/dialer/dialogs/SelectContactDialog.kt [hunk 2]
```diff
@@ -23,28 +31,115 @@ class SelectContactDialog(val activity: SimpleActivity, contacts: MutableList<Co
...
+        post {
+            setImageResource(iconResId)
+        }
+    }
+
+    private fun filterContactListBySearchQuery(query: String) {
+        val adapter = view.select_contact_list.adapter as? ContactsAdapter
+        var contactsToShow = contacts
+        if (query.isNotEmpty()) {
+            contactsToShow = contacts.filter { it.name.contains(query, true) }
```
Comment (Merkost) [inline app/src/main/kotlin/com/simplemobiletools/dialer/dialogs/SelectContactDialog.kt:112]: Do we need to add some sorting here? 
Just filtering by search string works well. Same in File Manager

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/dialer/dialogs/SelectContactDialog.kt:112]: use the already existing sorting, just filter out some items

**Comment by tibbi** [comment]
dialog is too high if no items are found, it shouldnt be. Cancel should be visible without having to close the keyboard

**Comment by Merkost** [comment]
> dialog is too high if no items are found, it shouldnt be. Cancel should be visible without having to close the keyboard

Yep, the invisible fastscroller thumb was there causing the big height, resolved that. 
Btw, do we need to use fastscroller in the dialog here? Haven't seen it used it in a dialog anywhere in the app

**Comment by tibbi** [comment]
of course, it is practical to use. Anyway, look good now, thanks

**Pull request merged**